src/gimelstudio/core/project_file.py

```diff
@@ -86,52 +86,56 @@ def SaveNodesData(self, nodes):
         self.contents["node_graph"]["nodes"] = nodes_data
 
     def CreateNodesFromData(self, nodegraph):
-        # Clear previous nodes and wires
-        nodegraph.nodes = {}
-        nodegraph.wires = []
+        try:
+            # Clear previous nodes and wires
+            nodegraph.nodes = {}
+            nodegraph.wires = []
 
-        # Create the nodes from the data
-        nodes_data = self.contents["node_graph"]["nodes"]
-        for node_id in nodes_data:
-            node_data = nodes_data[node_id]
+            # Create the nodes from the data
+            nodes_data = self.contents["node_graph"]["nodes"]
+            for node_id in nodes_data:
+                node_data = nodes_data[node_id]
 
-            # Create the node object
-            node = nodegraph.AddNode(node_data["idname"], 
+                # Create the node object
+                node = nodegraph.AddNode(node_data["idname"], 
                                     pos=node_data["pos"], 
                                     nodeid=node_id,
                                     location="POSITION")
```

Comment:
Code style: indent these lines
```suggestion
                node = nodegraph.AddNode(node_data["idname"], 
                                         pos=node_data["pos"], 
                                         nodeid=node_id,
                                         location="POSITION")
```